tests/test_async_downloader.py

```
@@ -0,0 +1,1401 @@
+
+
+# =============================================================================
+# Progress Callback Tests (lines 208-229)
+# =============================================================================
+
+
+@pytest.mark.asyncio
+class TestCallProgressCallback:
+    """Test _call_progress_callback method."""
+
+    async def test_sync_callback(self, mocker):
```

Ruff (0.15.0)

[warning] Unused method argument: `mocker`

(ARG002)